contracts/strategies/direct-grants-simple/DirectGrantsSimpleStrategy.sol

```diff
@@ -0,0 +1,415 @@
+        if (internalStatus == InternalRecipientStatus.InReview) {
+            return RecipientStatus.Pending;
+        } else {
+            return RecipientStatus(uint8(internalStatus));
+        }
+    }
+
+    /// @notice Returns the payout summary for the accepted recipient
+    function getPayouts(address[] memory _recipientIds, bytes memory, address)
+        external
+        view
+        returns (PayoutSummary[] memory payouts)
+    {
+        uint256 recipientLength = _recipientIds.length;
+
+        payouts = new PayoutSummary[](recipientLength);
+
+        for (uint256 i = 0; i < recipientLength;) {
+            payouts[i] = PayoutSummary(_recipientIds[i], _recipients[_recipientIds[i]].grantAmount);
+            unchecked {
+                i++;
+            }
+        }
+    }
+
+    /// @notice Checks if address is elgible allocator
+    /// @param _allocator Address of the allocator
+    function isValidAllocator(address _allocator) external view returns (bool) {
+        return allo.isPoolManager(poolId, _allocator);
+    }
+
+    /// @notice Get the status of the milestone of an recipient
+    /// @param _recipientId Id of the recipient
+    /// @param _milestoneId Id of the milestone
+    function getMilestoneStatus(address _recipientId, uint256 _milestoneId) public view returns (RecipientStatus) {
+        return milestones[_recipientId][_milestoneId].milestoneStatus;
+    }
+
+    /// ===============================
+    /// ======= External/Custom =======
+    /// ===============================
+
+    /// @notice Set milestones for recipient
+    /// @param _recipientId Id of the recipient
+    /// @param _milestones The milestones to be set
+    function setMilestones(address _recipientId, Milestone[] memory _milestones) external onlyPoolManager(msg.sender) {
+        if (upcomingMilestone[_recipientId] != 0) {
+            revert MILESTONES_ALREADY_SET();
+        }
+
+        uint256 totalAmountPercentage;
+
+        uint256 milestonesLength = _milestones.length;
+        for (uint256 i = 0; i < milestonesLength;) {
+            totalAmountPercentage += _milestones[i].amountPercentage;
+            milestones[_recipientId].push(_milestones[i]);
+
+            unchecked {
+                i++;
+            }
+        }
+
+        if (totalAmountPercentage != 1e18) {
+            revert INVALID_MILESTONE();
+        }
+
+        emit MilestonesSet(_recipientId);
+    }
+
+    /// @notice Submit milestone by the recipient
+    /// @param _recipientId Id of the recipient
+    /// @param _metadata The proof of work
+    function submitMilestone(address _recipientId, uint256 _milestoneId, Metadata calldata _metadata) external {
+        if (_recipientId != msg.sender && !_isIdentityMember(_recipientId, msg.sender)) {
+            revert UNAUTHORIZED();
+        }
+
+        Milestone[] storage recipientMilestones = milestones[_recipientId];
+
+        if (_milestoneId > recipientMilestones.length) {
```

Comment:
should we also verify that it fails if `_milestoneId < upcomingMilstone` ?

or is the check equally to 
```if (milestone.milestoneStatus == RecipientStatus.Accepted) {
            revert MILESTONE_ALREADY_ACCEPTED();
        }```

Reply:
I believe that would be addressed by the Accepted check